src/classes/mech/ActiveState.ts

```diff
@@ -369,40 +373,44 @@ class ActiveState {
   }
 
   RepairHP(): void {
-    this._mech.CurrentHP = this._mech.MaxHP
-    this._mech.CurrentRepairs -= 1
+    this.ActiveMech.CurrentHP = this.ActiveMech.MaxHP
+    this.ActiveMech.CurrentRepairs -= 1
   }
 
   RepairStructure(): void {
-    this._mech.CurrentStructure += 1
-    const cheap = this._mech.Bonuses.some(x => x.ID === 'cheap_struct')
-    this._mech.CurrentRepairs -= cheap ? 1 : 2
+    this.ActiveMech.CurrentStructure += 1
+    const cheap = this.ActiveMech.Bonuses.some(x => x.ID === 'cheap_struct')
+    this.ActiveMech.CurrentRepairs -= cheap ? 1 : 2
   }
 
   RepairStress(): void {
-    this._mech.CurrentStress += 1
-    if (this._mech.CurrentStress > this._mech.MaxStress)
-      this._mech.CurrentStress = this._mech.MaxStress
-    const cheap = this._mech.Bonuses.some(x => x.ID === 'cheap_stress')
-    this._mech.CurrentRepairs -= cheap ? 1 : 2
+    this.ActiveMech.CurrentStress += 1
+    if (this.ActiveMech.CurrentStress > this.ActiveMech.MaxStress)
+      this.ActiveMech.CurrentStress = this.ActiveMech.MaxStress
+    const cheap = this.ActiveMech.Bonuses.some(x => x.ID === 'cheap_stress')
+    this.ActiveMech.CurrentRepairs -= cheap ? 1 : 2
   }
 
   RepairSystem(w: MechEquipment): void {
     w.Repair()
-    this._mech.CurrentRepairs -= 1
+    this.ActiveMech.CurrentRepairs -= 1
   }
 
   RepairDestroyed(selfRepairPts: number): void {
-    this._mech.CurrentRepairs -= selfRepairPts
-    this._mech.Repair()
+    this.ActiveMech.CurrentRepairs -= selfRepairPts
+    this.ActiveMech.Repair()
   }
 
   public set ActiveMech(mech: Mech | null) {
-    this._mech = mech
+    if (this._pilot.IsLocallyOwned) {
+      this.ActiveMech = mech
+    } else {
+      this._remoteMech = mech
+    }
```

Comment:
There is a sed bug here, but this is the place where we differentiate based on locally owned status

Reply:
The bug is gone now